lib/dry/validation/extensions/struct.rb

```diff
@@ -0,0 +1,32 @@
+require 'dry-struct'
+
+module Dry
+  module Validation
+    class Schema
+      module StructClassBuilder
```

Comment:
@flash-gordon nvm, I need to understand the code better. My initial implementation had something like 
```ruby 
class Dry::Schema::Struct < Dry::Struct 
... 
end 
```